### PR TITLE
Enable dask resampling

### PR DESCRIPTION
**Pull request by BSchilperoort**
This PR modifies the xarray resampling routine to allow for parallelized dask computation.

A quick guide on getting started with Lilio + Dask is [added to the documentation](https://lilio--52.org.readthedocs.build/en/52/dask_integration.html). Linking to this from the resampling notebook would be nice, but currently the link would be dead (until this PR is merged).

### CODE REVIEW

**Comment by BSchilperoort** [comment]
Hey @geek-yang , would you want to go over the guide & try it out? 

I do still want to add tests for dask, however this will require generating some dummy netcdf files.

**Comment by geek-yang** [comment]
> Hey @geek-yang , would you want to go over the guide & try it out?
> 
> I do still want to add tests for dask, however this will require generating some dummy netcdf files.

Ok. Let me play with it first then. Thanks for the nice work 💯 !

**Review by geek-yang — COMMENTED** [review]
Fancy work @BSchilperoort ! Looks super awesome. I tested the code with some ERA5 data and it works well on my machine (except that my machine sounds like a big engine 😂 ).

I checked the doc and it works super nicely for me. But actually I'm still thinking it would be even nicer to have a notebook for it, which includes all the steps (e.g. create a calendar, map to data, etc). For me I know what to do to use calendar with it, but for new users I guess they need to follow the calendar notebook first and then they can move on with the instructions in the markdown. It would be nice to wrap everything up in one notebook.

Besides, I think we should emphasize it in the readme as well. We can add a line on top stressing that we support dask, which makes everything faster. If you want to learn the trick, go to link_to_md. I was also thinking about adding a dask badge (maybe a bit weird)?

Anyway, the code is already super awesome. Let's first write some tests and merge it. Then we can make a new release. For readme and notebook, we can work on it later.

Again, it wins my heart ❤️ ! Thanks for your effort @BSchilperoort.

File: lilio/resampling.py [hunk 1]
```diff
@@ -194,50 +194,46 @@ def _resample_dataset(
...
+    data_list = [xr.Dataset] * len(intervals)
+    for i in range(len(intervals)):
+        data_list[i] = xr.apply_ufunc(
+            resampling_method,
+            input_data_time.isel(time=indices_list[i]),
+            input_core_dims=[["time"]],
+            output_core_dims=[[]],
+            vectorize=True,
+            dask="parallelized",
+            dask_gufunc_kwargs={"allow_rechunk": True},
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply lilio/resampling.py]: Well, they just do nothing. I have added a comment here though.

**Comment by geek-yang** [comment]
> Thanks for the review! I am not sure about a notebook, as configuring for dask is quite specific to the use-case, and we would need to have example data stored somewhere, as well as more complicated docs dependencies.
> 
> Currently I have added a hyperlink to the docs page on Dask. The docs page is also right next to the example notebooks.

Ok sounds good. I think I can put a nice example in the recipe (I need to use it by any means) as indeed dask is specific and only for some advanced use cases with huge dataset. That's should be enough for now.

**Pull request merged**